src/uu/tail/src/tail.rs

```diff
@@ -214,39 +243,51 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
     } else {
         let multiple = files.len() > 1;
         let mut first_header = true;
-        let mut readers = Vec::new();
+        let mut file_objs = Vec::new();
 
         for filename in &files {
+            let path = Path::new(filename);
+            if path.is_dir() {
```

Comment:
please add some comments explaining the why here
maybe won't be needed if file_objs is named correctly